Fix CST parameterization method

Zcaic opened the pull request:
## Related Work
1. I add some code about **CST** to CSurfaceMovement.cpp and it can work well.

2. I add a test case named "cst" in **TestCases/deformation**. It contains test case for CST mesh deformation.



## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [ ] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I used the pre-commit hook to prevent dirty commits and used `pre-commit run --all` to format old commits.
- [x] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

pcarruscag commented on TestCases/deformation/cst/naca0012.cfg:
```diff
@@ -0,0 +1,133 @@
+% ------------------------- INPUT/OUTPUT FILE INFORMATION --------------------------%
+%
+% Mesh input file
+MESH_FILENAME= naca0012.cgns
```
```suggestion
MESH_FILENAME= ../naca0012/mesh_NACA0012_inv.su2
```

pcarruscag reviewed (commented):
Thanks, add yourself to AUTHORS.md please, since you added it as a testcase, we can try to make a regression test.
Can you check if your config works with the mesh I'm suggesting below (it's in the TestCases repo)? Then you need to change parallel_regression.py (look for /deformation/ to find the other tests).

Zcaic commented:
@pcarruscag ,
I have modified the **cfg** file and it passed the test with ```mpirun -n 2 SU2_DEF naca0012.cfg```.
I also modified **AUTHORS.md** and **parallel_regression.py**

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -1603,17 +1603,17 @@ def main():
-
-    pass_list.append(naca0012_def.run_def())
-    test_list.append(naca0012_def)
+    naca0012_cst            = TestCase('naca0012_def')
+    naca0012_cst.cfg_dir   = "deformation/naca0012"
+    naca0012_cst.cfg_file  = "def_NACA0012.cfg"
+    naca0012_cst.test_iter = 10
+    naca0012_cst.test_vals = [0.00352488] #residual
+    naca0012_cst.command   = TestCase.Command("mpirun -n 2", "SU2_DEF")
+    naca0012_cst.timeout   = 1600
+    naca0012_cst.tol       = 1e-8
+
+    pass_list.append(naca0012_cst.run_def())
+    test_list.append(naca0012_cst)
```
```suggestion
    naca0012_def            = TestCase('naca0012_def')
    naca0012_def.cfg_dir   = "deformation/naca0012"
    naca0012_def.cfg_file  = "def_NACA0012.cfg"
    naca0012_def.test_iter = 10
    naca0012_def.test_vals = [0.00352488] #residual
    naca0012_def.command   = TestCase.Command("mpirun -n 2", "SU2_DEF")
    naca0012_def.timeout   = 1600
    naca0012_def.tol       = 1e-8

    pass_list.append(naca0012_def.run_def())
    test_list.append(naca0012_def)
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -1771,6 +1771,21 @@ def main():
+    pass_list.append(naca0012_cst.run_def())
+    test_list.append(naca0012_cst)
+
+
+
```
```suggestion
```

pcarruscag commented on TestCases/deformation/cst/Mesh Deformation Comparison.png:
This file is a bit large, use a smaller resolution or remove please.

pcarruscag commented on Common/src/grid_movement/CSurfaceMovement.cpp:
```diff
@@ -584,6 +585,16 @@ vector<vector<su2double> > CSurfaceMovement::SetSurface_Deformation(CGeometry* g
       }
     }
 
+     /*--- Apply the design variables to the control point position ---*/
```
```suggestion
    /*--- Apply the design variables to the control point position ---*/
```

pcarruscag commented on AUTHORS.md:
```diff
@@ -160,4 +160,5 @@ sravya91
 srcopela
 tobadavid
 vfrancesmolla
+Zcaic
```
Because of how Git outputs this list, names with upper case come first, move to after "Zan Xu" please.

pcarruscag commented on Common/src/grid_movement/CSurfaceMovement.cpp:
```diff
@@ -586,6 +587,16 @@ vector<vector<su2double> > CSurfaceMovement::SetSurface_Deformation(CGeometry* g
     /*--- Apply the design variables to the control point position ---*/
 
+    for (iDV = 0; iDV < config->GetnDV(); iDV++) {
+      switch (config->GetDesign_Variable(iDV)) {
+        case CST:
+          SetCST(geometry, config, iDV, false);
+          break;
+      }
```
```suggestion
      if (config->GetDesign_Variable(iDV) == CST) {
        SetCST(geometry, config, iDV, false);
      }
```

Zcaic commented:
Sorry, some was caused by my carelessness,. I've modified it.

pcarruscag commented on TestCases/deformation/cst/naca0012.cfg:
```diff
@@ -0,0 +1,133 @@
+DV_VALUE= -0.5,0.5
+
+% ------------------------- INPUT/OUTPUT FILE INFORMATION --------------------------%
+%
```
```suggestion
%
DEFORM_CONSOLE_OUTPUT= YES
%
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -1771,6 +1771,19 @@ def main():
+    naca0012_cst.cfg_dir   = "deformation/cst"
+    naca0012_cst.cfg_file  = "naca0012.cfg"
+    naca0012_cst.test_iter = 10
+    naca0012_cst.test_vals = [0.00352488] #residual
```
```suggestion
    naca0012_cst.test_vals = [0.000386] #residual
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -1771,6 +1771,19 @@ def main():
+    naca0012_cst.cfg_dir   = "deformation/cst"
+    naca0012_cst.cfg_file  = "naca0012.cfg"
+    naca0012_cst.test_iter = 10
+    naca0012_cst.test_vals = [0.000386] #residual
```
```suggestion
    naca0012_cst.test_vals = [0.000385514] #residual
```
